prac_07/project_management.py

```diff
@@ -0,0 +1,113 @@
+        "(U)pdate project\n(Q)uit")
+FILENAME = "projects.txt"
+projects = []
+
+
+def main():
+    print(MENU)
+    choice = input(">>>: ").upper()
+    while choice != "Q":
+        if choice == "L":
+            load_projects(FILENAME)
+        elif choice == "S":
+            save_projects(FILENAME)
+        elif choice == "D":
+            display_projects()
+        elif choice == "F":
+            filter_projects()
+        elif choice == "A":
+            add_projects()
+        elif choice == "U":
+            update_project()
+        else:
+            print("Invalid choice")
+        print(MENU)
+        choice = input(">>>: ").upper()
+    print("Thank you for using custom-built project management software.")
+
+
+def add_projects():
+    """Add new projects"""
+    print(f"Let's add a new project")
+    name = input("Name: ")
+    start_date = input("Start date (dd/mm/yy): ")
+    start_date = datetime.datetime.strptime(start_date, "%d/%m/%Y").date()
+    priority = int(input("Priority: "))
+    cost_estimate = float(input("Cost estimate: $"))
+    completion_percentage = int(input("Percent complete: "))
+    project_to_add = Project(name, start_date, priority, cost_estimate, completion_percentage)
+    projects.append(project_to_add)
+
+
+def filter_projects():
+    """Filter and display projects that start after the input date."""
+    filtered_date = input("Show projects that start after date (dd/mm/yy): ")
+    filtered_date = datetime.datetime.strptime(filtered_date, "%d/%m/%Y").date()
+    filtered_projects = []
+    for project in projects:
+        if project.start_date >= filtered_date:
+            filtered_projects.append(project)
+    filtered_projects.sort()
+    for filtered_project in filtered_projects:
+        print(filtered_project)
+
+
+def display_projects():
+    """Display sorted incompleted projects and completed projects nicely"""
+    incompleted_projects = [project for project in projects if int(project.completion_percentage) < 100]
+    incompleted_projects.sort()
+    print("Incomplete projects: ")
+    for incompleted_project in incompleted_projects:
+        print(f"  {incompleted_project}")
+    completed_projects = [project for project in projects if int(project.completion_percentage) == 100]
+    completed_projects.sort()
+    print("Complete projects: ")
+    for completed_project in completed_projects:
+        print(f"  {completed_project}")
+
+
+def save_projects(filename):
+    """Save update projects into projects.txt"""
+    with open(filename, "w") as out_file:
+        out_file.write("Name\tStart Date\tPriority\tEstimate\tCompletion\n")
+        for project in projects:
+            out_file.write(f"{project.name}\t{project.start_date.strftime('%d/%m/%Y')}\t{project.priority}"
+                           f"\t{project.cost_estimate:.2f}\t{project.completion_percentage}\n")
+
+
+def load_projects(filename):
```

Comment:
I think you should allow user input the project name as the requirement:
Load projects
(Prompt the user for a filename to load projects from and load them)

Reply:
oki thanks